### DISCUSSION TITLE
Feature discussion: 0 js by default

### DISCUSSION THREAD

**Discussion by orenelbaum** [solidjs/solid-start#173] [Ideas]
Also known as "removing the script tag".
We see this in frameworks like Astro, Fresh, and I think Remix.
I think that it has some tradeoffs. It's obviously better to ship no js than js that doesn't do anything.
On the other hand I don't know if we want to encourage people to ever ship no js, I think that almost every website could be more accessible with at least some js.
I'd probably still ship this feature cause in terms of real tradeoffs it's hard to say what's better, but the hype factor is very big for this feature. I'm not sure how hard it is to implement but it doesn't sound like it should be too hard

**Comment by ryansolid** [comment]
Have you tried removing the `<Scripts>` tag?

I'm basically saying I think we already support this.

**Reply by orenelbaum** [reply]
Would it be hard to automate it? I don't know if it will really bring value I'm just curious why other frameworks have it.

**Reply by ryansolid** [reply]
I mean Remix doesn't. You remove the `<Scripts>` tag. Or render it conditionally. SvelteKit marks it on the page meta data. But not default for any of these.

Astro, Fresh, Marko, Qwik all are MPA frameworks with partial hydration so it is automated because it's the default. A SPA framework wouldn't ever make this the default since it is all or nothing at a page level. No way to be incremental here.

**Reply by orenelbaum** [reply]
Oh I see, that makes sense. When you think about it, it actually makes more sense in the context of partial hydration.

**Comment by boehs** [comment]
We support islands now, which should theoretically (correct me if I'm wrong @nksaraf) introduce fully opt-in javascript.

See #203 #190

**Reply by ryansolid** [reply]
Experimental. Islands will not be considered stable at beta release time. But yes this would.
